File: springboot-backend/src/main/java/com/wedspot/backend/Model/Entity/Vendor.java
package com.wedspot.backend.Model.Entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "vendors")
public class Vendor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String businessName;
    private String description;
    private String category;
    private String priceRange;
    private String location;
    private Double rating = 0.0;
    private Integer reviewCount = 0;
    private String imageUrl;
    
    private boolean verified = false;
    private boolean premium = false;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: backend/Models/Entities/Vendor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models.Entities
{
    [Table("vendors")]
    public class Vendor
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        public string BusinessName { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string? Category { get; set; }

        public string? PriceRange { get; set; }

        public string? Location { get; set; }

        public double Rating { get; set; } = 0.0;

        public int ReviewCount { get; set; } = 0;

        public string? ImageUrl { get; set; }

        public bool Verified { get; set; } = false;

        public bool Premium { get; set; } = false;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
